packages/laconia-core/src/index.d.ts

```diff
@@ -20,7 +20,8 @@ declare namespace laconia {
 
   interface LaconiaHandler extends Handler {
     register(
-      factory: LaconiaFactory | LaconiaFactory[],
+      factory: string | LaconiaFactory | LaconiaFactory[],
```

Comment:
TypeScript supports overloading. You should be able to do something like this:

```
  interface LaconiaHandler extends Handler {
    register(
      factory: LaconiaFactory | LaconiaFactory[],
      options?: FactoryOptions
    ): LaconiaHandler;
    register(
      instanceName: string,
      factory: LaconiaFactory,
      options?: FactoryOptions
    ): LaconiaHandler;
  }
```

Reply:
I like this, but I can't seem to get it to pass. Do the arguments in the typescript definition need to match the argument names of the function?

Reply:
Sorry, I can't push for changes as it's in your fork instead of a branch. The below code has been tested locally and it seems to work for me. Otherwise, do merge even without this change, and I'll make the change to the TS files.

laconia-core index.d.ts
```ts
  interface LaconiaHandler extends Handler {
    register(
      factory: LaconiaFactory | LaconiaFactory[],
      options?: FactoryOptions
    ): LaconiaHandler;
    register(
      name: string,
      factory: LaconiaFactory,
      options?: FactoryOptions
    ): LaconiaHandler;
  }
```

laconia-batch index.d.ts
```ts
interface LaconiaBatchHandler extends LaconiaHandler {
  on(
    eventName: "start" | "end",
    eventListener: BatchEventListener
  ): LaconiaBatchHandler;
  on(eventName: "item", eventListener: ItemEventListener): LaconiaBatchHandler;
  on(eventName: "stop", eventListener: StopEventListener): LaconiaBatchHandler;
  register(
    factory: LaconiaFactory | LaconiaFactory[],
    options?: FactoryOptions
  ): LaconiaBatchHandler;
  register(
    name: string,
    factory: LaconiaFactory,
    options?: FactoryOptions
  ): LaconiaBatchHandler;
}
```

Reply:
Ah I was missing the one in laconia-batch, thanks!